autorest/azure/async.go

```diff
@@ -444,24 +446,26 @@ func (pt *pollingTrackerBase) updateErrorFromResponse() {
 		re := respErr{}
 		defer pt.resp.Body.Close()
 		var b []byte
-		if b, err = ioutil.ReadAll(pt.resp.Body); err != nil {
-			goto Default
-		}
-		if err = json.Unmarshal(b, &re); err != nil {
-			goto Default
-		}
-		// unmarshalling the error didn't yield anything, try unwrapped error
-		if re.ServiceError == nil {
-			err = json.Unmarshal(b, &re.ServiceError)
-			if err != nil {
+		if len(b) != 0 {
```

Comment:
This is incorrect, you're checking the length before reading from the body.

Reply:
Oops. You are right. Moved if statement to the right place